keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandlerTests.kt

```diff
@@ -190,21 +194,52 @@ internal object ClusterHandlerTest : JUnit5Minutests {
         }
       }
 
-      test("annealing a diff clones the current server group") {
-        whenever(orcaService.orchestrate(any())) doReturn CompletableDeferred(TaskRefResponse("/tasks/1"))
+      context("the diff is only in capacity") {
 
-        converge(request.metadata.name, spec)
+        val diff = javers.compare(resource.spec.withDoubleCapacity(), resource.spec)
 
-        argumentCaptor<OrchestrationRequest>().apply {
-          verify(orcaService).orchestrate(capture())
+        test("annealing resizes the current server group") {
+          upsert(resource, diff)
+
+          argumentCaptor<OrchestrationRequest>().apply {
+            verify(orcaService).orchestrate(capture())
+
+            expectThat(firstValue.job.first()) {
+              get("type").isEqualTo("resizeServerGroup")
+              get("capacity").isEqualTo(
+                mapOf(
+                  "min" to spec.capacity.min,
+                  "max" to spec.capacity.max,
+                  "desired" to spec.capacity.desired
+                )
+              )
+              get("serverGroupName").isEqualTo(activeServerGroupResponse.asg.autoScalingGroupName)
+            }
+          }
+        }
+      }
+
+      context("the diff is something other than just capacity") {
 
-          expectThat(firstValue.job.first()["source"])
-            .isA<Map<String, Any>>()
-            .and {
-              hasEntry("account", activeServerGroupResponse.accountName)
-              hasEntry("region", activeServerGroupResponse.region)
-              hasEntry("asgName", activeServerGroupResponse.asg.autoScalingGroupName)
+        val diff = javers.compare(resource.spec.withDoubleCapacity().withDifferentInstanceType(), resource.spec)
+
+        test("annealing clones the current server group") {
```

Comment:
Tests that if there are more changes than just capacity it does a "createServerGroup"